java-manta-it/src/test/java/com/joyent/test/util/TestListingInterceptor.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2019, Joyent, Inc. All rights reserved.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.joyent.test.util;
+
+import org.apache.commons.lang3.BooleanUtils;
+import org.apache.commons.lang3.ObjectUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testng.IMethodInstance;
+import org.testng.IMethodInterceptor;
+import org.testng.ISuite;
+import org.testng.ISuiteListener;
+import org.testng.ITestContext;
+import org.testng.ITestNGMethod;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentLinkedQueue;
+
+/**
+ * Test method listener for determining which tests will actually run.
+ *
+ * Enabled by: -Dit.dryRun=true
+ * @author <a href="https://github.com/nairashwin952013">Ashwin A Nair</a>
+ */
+public class TestListingInterceptor implements IMethodInterceptor, ISuiteListener {
+
+    private static final Logger LOG = LoggerFactory.getLogger(TestListingInterceptor.class);
+
+    private static final List<IMethodInstance> EMPTY_LIST = new ArrayList<>(0);
+
+    private static final ConcurrentLinkedQueue<String> observedTests = new ConcurrentLinkedQueue<>();
+
+    private static final boolean DRY_RUN_ENABLED;
+
+    static {
+        DRY_RUN_ENABLED = ObjectUtils.firstNonNull(
+                BooleanUtils.toBoolean(System.getProperty("it.dryRun")),
+                false);
+    }
+
+    public static int getObservedTestCount() {
+        return observedTests.size();
+    }
+
+    @Override
+    public List<IMethodInstance> intercept(final List<IMethodInstance> methods, final ITestContext context) {
```

Comment:
Please add a comment to describe what you are intending to do here.

Reply:
`javadoc` Added.